core/utils/svg_paths.js

```diff
@@ -0,0 +1,153 @@
+  return ' ' + x + ',' + y + ' ';
+};
+/**
+ * Draw a curbic or quadratic curve.  See
+ * developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d#Cubic_B%C3%A9zier_Curve
+ * These coordinates are unitless and hence in the user coordinate system.
+ * @param {string} command The command to use.
+ *     Should be one of: c, C, s, S, q, Q.
+ * @param {string} points  An array containing all of the points to pass to the
+ *     curve command, in order.  The points are represented as strings of the
+ *     format ' x, y '.
+ * @return {string} A string defining one or more Bezier curves.  See the MDN
+ *     documentation for exact format.
+ * @public
+ */
+Blockly.utils.Paths.curve = function(command, points) {
+  return command + points.join('');
+};
+/**
+ * Move the cursor to the given position without drawing a line.
+ * The coordinates are absolute.
+ * These coordinates are unitless and hence in the user coordinate system.
+ * See developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#Line_commands
+ * @param {number} x The absolute x coordinate.
+ * @param {number} y The absolute y coordinate.
+ * @return {string} A string of the format ' M x,y '
+ * @public
+ */
+Blockly.utils.Paths.moveTo = function(x, y) {
+  return ' M ' + x + ',' + y + ' ';
+};
+/**
+ * Move the cursor to the given position without drawing a line.
+ * Coordinates are relative.
+ * These coordinates are unitless and hence in the user coordinate system.
+ * See developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#Line_commands
+ * @param {number} dx The relative x coordinate.
+ * @param {number} dy The relative y coordinate.
+ * @return {string} A string of the format ' m dx,dy '
+ * @public
+ */
+Blockly.utils.Paths.moveBy = function(dx, dy) {
+  return ' m ' + dx + ',' + dy + ' ';
+};
+
+/**
+ * Draw a line from the current point to the end point, which is the current
+ * point shifted by dx along the x-axis and dy along the y-axis.
+ * These coordinates are unitless and hence in the user coordinate system.
+ * See developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#Line_commands
+ * @param {number} dx The relative x coordinate.
+ * @param {number} dy The relative y coordinate.
+ * @return {string} A string of the format ' L dx,dy '
+ * @public
+ */
+Blockly.utils.Paths.lineTo = function(dx, dy) {
+  return ' l ' + dx + ',' + dy + ' ';
+};
+
+/**
+ * Draw multiple lines connecting all of the given points in order.  This is
+ * equivalent to a series of 'l' commands.
+ * These coordinates are unitless and hence in the user coordinate system.
+ * See developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#Line_commands
+ * @param {!Array.<string>} points An array containing all of the points to
+ *     draw lines to, in order.  The points are represented as strings of the
+ *     format ' dx, dy '.
+ * @return {string} A string of the format ' l (dx,dy)+ '
+ * @public
+ */
+Blockly.utils.Paths.line = function(points) {
+  return 'l' + points.join('');
```

Comment:
Did you want to add spaces on either side of this, as specified in the JSDoc?

Reply:
Yes, though it's only needed at the beginning because the points have spaces.

This way of doing this does mean that there are some double spaces between things in the svg string.  I don't consider this a big issue.  Browsers don't care.